forward/forward.go

```diff
@@ -9,103 +9,73 @@ import (
 	"time"
 
 	"github.com/getlantern/golog"
-	"github.com/getlantern/http-proxy/utils"
 	"github.com/getlantern/idletiming"
+
+	"github.com/getlantern/http-proxy/filter"
 )
 
 var log = golog.LoggerFor("forward")
 
-type Forwarder struct {
-	errHandler   utils.ErrorHandler
-	roundTripper http.RoundTripper
-	rewriter     RequestRewriter
-	next         http.Handler
-
-	idleTimeout time.Duration
+type forwarder struct {
+	*Options
 }
 
-type optSetter func(f *Forwarder) error
-
-func RoundTripper(r http.RoundTripper) optSetter {
-	return func(f *Forwarder) error {
-		f.roundTripper = r
-		return nil
-	}
+type Options struct {
+	IdleTimeout  time.Duration
+	Rewriter     RequestRewriter
+	RoundTripper http.RoundTripper
 }
 
 type RequestRewriter interface {
 	Rewrite(r *http.Request)
 }
 
-func Rewriter(r RequestRewriter) optSetter {
-	return func(f *Forwarder) error {
-		f.rewriter = r
-		return nil
```

Comment:
Also, note that we `Forwarder` is now `forwarder` (un-exported) because the client doesn't actually need the concrete type since all the initialization has already been done :)